### PR TITLE
Add a Touch ID prompt message to `tsh`

### PR DESCRIPTION
**Pull request by codingllama**
Add a prompt message to `tsh` when we fire Touch ID, so users are aware of it. Helpful in certain situations where the prompt isn't apparent, such as partially open macbooks, multiple monitors, `tsh` being used from remote sessions, etc.

The Touch ID prompt is shown automatically by the OS in certain operations, which are triggered via native code, so the message itself is added before the first invocation of presence-requiring methods.

### CODE REVIEW

**Comment by codingllama** [comment]
Some prompt examples:

```shell
$ ./tsh login --proxy=zarquon --user=llama
> Using platform authenticator, follow the OS prompt
> ...

$ ./tsh mfa add --mfa-mode=cross-platform
> Choose device type [TOTP, WEBAUTHN, TOUCHID]: touchid
> Enter device name: tsh4
> Tap any *registered* security key
> Using platform authenticator, follow the OS prompt
> MFA device "tsh4" added.
# Note: 1st tap is for authn, 2nd tap is to add the touchid device.

$ ./tsh touchid ls
> Using platform authenticator, follow the OS prompt
> ...

$ ./tsh touchid rm 0fd5accc-7205-4881-a72a-5ba2e3d8a3e0
> Using platform authenticator, follow the OS prompt
> Touch ID credential "0fd5accc-7205-4881-a72a-5ba2e3d8a3e0" removed.
```

Happy to tweak the message. FYI @xinding33 @espadolini.

**Review by Tener — APPROVED** [review]

File: lib/auth/touchid/api.go [hunk 1]
```diff
@@ -42,14 +44,29 @@ import (
 var (
 	ErrCredentialNotFound = errors.New("credential not found")
 	ErrNotAvailable       = errors.New("touch ID not available")
+
+	// PromptPlatformMessage is the message shown before Touch ID prompts.
+	PromptPlatformMessage = "Using platform authenticator, follow the OS prompt"
+	// PromptWriter is the writer used for prompt messages.
+	PromptWriter io.Writer = os.Stderr
```
Comment (jakule) [inline lib/auth/touchid/api.go:51]: Do these variables need to be public?
Why do not make `PromptPlatformMessage` a const?

File: lib/auth/touchid/api.go [hunk 2]
```diff
@@ -42,14 +44,29 @@ import (
...
 )
 
+func promptPlatform() {
+	if PromptPlatformMessage != "" {
```
Comment (jakule) [inline lib/auth/touchid/api.go:55]: Why do we need this check?

**Review by jakule — APPROVED** [review]
A few questions, but apart from that LGTM.

File: lib/auth/touchid/api.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/touchid/api.go:51]: I made them public so they can be changed/customized from caller packages. We don't need to tweak those messages today, but we do so for some other prompt messages, so it's not a big stretch.

File: lib/auth/touchid/api.go (hunk 2 above)
Comment (codingllama) [reply lib/auth/touchid/api.go:55]: It's tied to the question above: if for some reason we want to disable the message, an outside package can set it to empty and the Println is skipped.

**Comment by codingllama** [comment]
Thanks folks for the review. I'll ping @xinding33 and @zmb3 for feedback, but might merge if we don't hear back in a timely manner: _a_ message is better than no message, I think.

**Review by zmb3 — APPROVED** [review]
No concerns here.

**Comment by xinding33** [comment]
Great idea. Prompt messages look good to me!

**Pull request merged**